### PR TITLE
feat(appeals): switching enums to the definitions generated in data-models

### PR DESCRIPTION
**Pull request by fabfab-kc**
All data enumerations now use the definitions generated in `pins-data-model`.
TODO: a couple of APPEAL_CASE_STAGE and APPEAL_DOCUMENT_TYPE entries ("internal" and "dropbox") have been hardcoded, will be aligned on the next data model update.

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/api/src/server/tests/appeals/expectation.js [hunk 1]
```diff
@@ -15,11 +15,11 @@ export const baseExpectedLPAQuestionnaireResponse = (appeal) => ({
 		// @ts-ignore
 		whoNotified: {},
 		// @ts-ignore
-		siteNotice: {},
+		whoNotifiedSiteNotice: {},
```
Comment (BiggsKC) [inline appeals/api/src/server/tests/appeals/expectation.js:18]: will this change affect my previous work where I use 'siteNotice' for LPAQ?

Comment (fabfab-kc) [reply appeals/api/src/server/tests/appeals/expectation.js:18]: The test was failing - this is why I changed them. I think once part 2 is done, will be aligned...

File: appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js [hunk 2]
```diff
@@ -15,8 +15,8 @@ import { surnameFirstToFullName } from '#lib/person-name-formatter.js';
 import { preRenderPageComponents } from '#lib/nunjucks-template-builders/page-component-rendering.js';
 import { addNotificationBannerToSession } from '#lib/session-utilities.js';
 import { redactionStatusIdToName } from '#lib/redaction-statuses.js';
-import { AVSCAN_STATUS } from '@pins/appeals/constants/documents.js';
-import { REDACTION_STATUS } from '@pins/appeals/constants/documents.js';
+import { APPEAL_VIRUS_CHECK_STATUS } from 'pins-data-model';
+import { APPEAL_REDACTED_STATUS } from 'pins-data-model';
```
Comment (BiggsKC) [inline appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js:19]: can be added to line above

File: appeals/web/src/server/lib/mappers/appeal.mapper.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ import {
 	mapDocumentDownloadUrl,
 	mapVirusCheckStatus
 } from '#appeals/appeal-documents/appeal-documents.mapper.js';
-import { AVSCAN_STATUS } from '@pins/appeals/constants/documents.js';
+import { APPEAL_VIRUS_CHECK_STATUS } from 'pins-data-model';
```
Comment (BiggsKC) [inline appeals/web/src/server/lib/mappers/appeal.mapper.js:19]: can be added to line below (2 pins-data-model imports)

File: appeals/web/testing/app/fixtures/referencedata.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import { REDACTION_STATUS } from '@pins/appeals/constants/documents.js';
-import { AVSCAN_STATUS } from '@pins/appeals/constants/documents.js';
+import { APPEAL_REDACTED_STATUS } from 'pins-data-model';
```
Comment (BiggsKC) [inline appeals/web/testing/app/fixtures/referencedata.js:1]: can be merged (2 pins-data-model imports)

File: appeals/api/src/server/tests/appeals/expectation.js (hunk 1 above)
Comment (BiggsKC) [reply appeals/api/src/server/tests/appeals/expectation.js:18]: api test failing? if so I couldn't have merged my branch.
We have 'siteNotice' etc in `openapi.json`, not `whoNotifiedSiteNotice`

**Review by BiggsKC — APPROVED** [review]
All good!

**Pull request merged**